src/aggregations/top-hits-aggregation.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal counter for added top hits aggregations.
+ * @type {Number}
+ */
+let count = 0;
+
+/**
+ * Construct a Top hits aggregation.
+ *
+ * @param  {Object} opts  Options to include in the aggregation (maybe empty).
+ * @param  {String} name  Aggregation name. Defaults to agg_top_hits_{count}.
+ * @return {Object}       Top hits Aggregation.
+ */
+export default function topHitsAggregation(opts, name) {
+  name = name || `agg_top_hits_${count++}`
```

Comment:
WFM to leave as-is. Technically all aggregations would need to maintain their own counters too, right? These naming conflicts are not unique to the top-hits aggregation AFAIK. That can be deferred to a future PR.